### ISSUE TITLE
ATOM_REPOS_HOME environment variable seems to be ignored.

### ISSUE THREAD

**Issue by danielbayley** [atom/package-generator#33]
Despite having specified the `ATOM_REPOS_HOME` environment variable (in my `~/.bashrc`), whenever I do `Generate Atom Package` I still just get the default `~/github/my-package`. Yet if I open the Developer Tools window and do `process.env.ATOM_REPOS_HOME` it gives me the correct path that I specified.

Even if I put

``` coffee
process.env.ATOM_REPOS_HOME = "#{process.env.ATOM_HOME}/dev/packages"
```

into `init.coffee` it still has no effect. It does actually work properly if I specify `projectHome` in my `config.cson` as a workaround, but it should respect the environment variable regardless.

It's strange; as I can see in `/lib/package-generator-view.coffee` the relevant code block should work…

``` coffee
getPackagesDirectory: ->
    atom.config.get('core.projectHome') or
      process.env.ATOM_REPOS_HOME or
      path.join(fs.getHomeDirectory(), 'github')
```

Any ideas?

**Comment by chrishiestand** [comment]
This is still a problem in atom 1.7.3

**Comment by chrishiestand** [comment]
nevermind, sorry. I was having trouble with `apm dev` installing to the correct location and I mistakenly thought it was the same issue. Please ignore my comment unless someone else can reproduce the bug.

**Comment by thomasjo** [comment]
@danielbayley Is this still a problem for you? What does `atom.config.get('core.projectHome')` produce if you run that in the developer tools console? As you can see from the relevant code block, the environment variable `ATOM_REPOS_HOME` is only used if `core.projectHome` is **not** set.

**Comment by danielbayley** [comment]
I have `core.projectHome` setup correctly, but seem to remember there been a problem if it isn't set even if `ATOM_REPOS_HOME` _is_. I'll have another look into this issue when I get chance.

What is the difference/purpose between `ATOM_REPOS_HOME` and `ATOM_DEV_RESOURCE_PATH`?

**Comment by thomasjo** [comment]
The latter is where the path to the Atom source code, and is used when live reloading Atom in dev mode. It allows us to make changes to the core Atom code and test those changes without needing to run e.g. `script/build`.

As for your comment about `core.projectHome` being setup correctly; if you've set it to a value, then `ATOM_REPOS_HOME` will never be used, regardless of how you set it. The value of `core.projectHome` will always win unless it's set to a "falsey" value (e.g. empty string, `null`, `false`, and similar).

**Comment by danielbayley** [comment]
@thomasjo Iv'e just been looking at _package-generator_ again —thinking about how it might fit in with my plans for a more general templating system— and the problem is actually a simple logic mismatch between this and the apm package…

I set `ATOM_REPOS_HOME` so that `apm dev` will clone where I want, and then leave `core.projectHome` unset because I want to rely on the same variable, but the problem is that in this package `core.projectHome` is not 'falsey' (and so _can't_ make use of `ATOM_REPOS_HOME`) because it is set to the default value of `~/github`.

So I think the solution would be to swap lines [80] and [81] so that `ATOM_REPOS_HOME` takes priority. What do you think?

[80]: https://github.com/atom/package-generator/blob/master/lib/package-generator-view.coffee#L80
[81]: https://github.com/atom/package-generator/blob/master/lib/package-generator-view.coffee#L81

**Comment by ecwpz91** [comment]
Added the following to `init.coffee`.

```sh
atom.workspace.packageManager.config.defaultSettings.core.projectHome = '/your/path/here'
```

**Comment by 50Wliu** [comment]
Believe this has been fixed by #50.
